crates/noirc_evaluator/src/brillig/brillig_ir/debug_show.rs

```diff
@@ -130,215 +127,313 @@ impl<T: DebugToString> DebugToString for [T] {
 }
 
 macro_rules! debug_println {
-    ( $literal:expr ) => {
-        if ENABLE_DEBUG_TRACE {
-            println!("{}", $literal);
+    ( $enable_debug:expr, $literal:expr ) => {
+        match $enable_debug {
+            true => println!("{}", $literal),
+            false => ()
         }
     };
-    ( $format_message:expr, $( $x:expr ),* ) => {
-        if ENABLE_DEBUG_TRACE {
-            println!($format_message, $( $x.debug_to_string(), )*)
+    ( $enable_debug:expr, $format_message:expr, $( $x:expr ),* ) => {
+        match $enable_debug {
+            true => println!($format_message, $( $x.debug_to_string(), )*),
+            false => ()
```

Comment:
These should probably be `if`s instead of matches.

Reply:
I switched to if statement and it dismissed your review if you could approve once more @jfecher